src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-form/rbd-form.component.html

```diff
@@ -0,0 +1,303 @@
+    <li class="breadcrumb-item">Block</li>
+    <li class="breadcrumb-item">
+      <a routerLink="/block/rbd">Images</a></li>
+    <li class="breadcrumb-item active">{{ editing ? 'Edit' : 'Add' }}</li>
+  </ol>
+</nav>
+
+<div class="col-sm-12 col-lg-6">
+  <form name="rbdForm"
+        class="form-horizontal"
+        #formDir="ngForm"
+        [formGroup]="rbdForm"
+        novalidate>
+    <div class="panel panel-default">
+      <div class="panel-heading">
+        <h3 class="panel-title">{{ editing ? 'Edit' : 'Add' }} RBD</h3>
+      </div>
+      <div class="panel-body">
+
+        <!-- Name -->
+        <div class="form-group"
+             [ngClass]="{'has-error': (formDir.submitted || rbdForm.controls.name.dirty) && rbdForm.controls.name.invalid}">
+          <label i18n
+                 class="control-label col-sm-3"
+                 for="name">Name
+            <span class="required"></span>
+          </label>
+          <div class="col-sm-9">
+            <input class="form-control"
+                   type="text"
+                   placeholder="Name..."
+                   id="name"
+                   name="name"
+                   formControlName="name"
+                   autofocus>
+            <span i18n
+                  class="help-block"
+                  *ngIf="(formDir.submitted || rbdForm.controls.name.dirty) && rbdForm.controls.name.hasError('required')">
+              This field is required.
+            </span>
+          </div>
+        </div>
+
+        <!-- Pool -->
+        <div class="form-group"
+             [ngClass]="{'has-error': (formDir.submitted || rbdForm.controls.pool.dirty) && rbdForm.controls.pool.invalid}"
+             (change)="onPoolChange($event.target.value)">
+          <label class="control-label col-sm-3"
+                 for="pool">
+            Pool
+            <span class="required"></span>
+          </label>
+          <div class="col-sm-9">
+            <select id="pool"
+                    name="pool"
+                    class="form-control"
+                    formControlName="pool">
+              <option *ngIf="pools === null"
+                      [ngValue]="null">Loading...
+              </option>
+              <option *ngIf="pools !== null && pools.length === 0"
+                      [ngValue]="null">-- No pools available --
+              </option>
+              <option *ngIf="pools !== null && pools.length > 0"
+                      [ngValue]="null">-- Select a pool --
+              </option>
+              <option *ngFor="let pool of pools"
+                      [value]="pool.pool_name">{{ pool.pool_name }}</option>
+            </select>
+            <span i18n
+                  class="help-block"
+                  *ngIf="(formDir.submitted || rbdForm.controls.pool.dirty) && rbdForm.controls.pool.hasError('required')">
+              This field is required.
+            </span>
+          </div>
+        </div>
+
+        <!-- Use a dedicated pool -->
```

Comment:
You should not show this option if no data pool can be selected.

Reply:
From the usability point of view, I'm in favor of not hiding this field.

If no data pool can be selected user will see a "-- No data pool available --" on the dropdown.